File: baekjoon/bj11478.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.Set;

public class bj11478 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String str = br.readLine();

        Set<String> strarr = new HashSet<String>();
        for (int i = 0; i < str.length(); i++) {
            for (int j = i + 1; j < str.length() + 1; j++) {
                strarr.add(str.substring(i, j));
            }
        }

        System.out.println(strarr.size());

    }
}

File: baekjoon/bj11478.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main(){
    val br= BufferedReader(InputStreamReader(System.`in`))
    val str= br.readLine()
    val nameSet=mutableSetOf<String>()
    for (start in 0..str.length){
        for(end in start+1..str.length){
            nameSet.add(str.substring(start,end))
        }
    }
    println(nameSet.size)

}